// COMPARE_WITH_LIGHT_TREE

enum class E {
    ENTRY;

    fun <!VIRTUAL_MEMBER_HIDDEN!>getDeclaringClass<!>() {}
    <!ACCIDENTAL_OVERRIDE{LT}!><!ACCIDENTAL_OVERRIDE{PSI}!>fun <!VIRTUAL_MEMBER_HIDDEN!>finalize<!>()<!> {}<!>
}
